docworks-dts/lib/multiple-files/convertors.js

```diff
@@ -0,0 +1,228 @@
+const { resolveType } = require('./typeResolver')
+const {
+	dtsEnum,
+	dtsConst,
+	dtsMethod,
+	dtsFunction,
+	dtsParameter,
+	dtsProperty,
+	dtsInterface,
+	dtsNamespace,
+	dtsObjectTypeAlias,
+	dtsFunctionTypeAlias
+} = require('./generators')
+
+const getDocumentationGenerator = require('./providers/documentationGenerator')
+const { getServiceSummary } = require('./serviceUtils')
+
+const { SUB_SERVICES_KEY } = require('./constants')
+
+const propertyToConst = (property, service) => {
+	const documentationGenerator = getDocumentationGenerator()
+	const jsDocComment = documentationGenerator({
+		summary: property.docs.summary,
+		service: service.name,
+		member: property.name
+	})
+	property.type = resolveType(property.type, service, { union: true })
+	return dtsConst(property, { jsDocComment, service })
+}
+
+const operationParamToParameters = (param, service) => {
+	const { name, type, optional, doc, spread } = param
+	const paramType = spread ? { name: 'Array', typeParams: [type] } : type
+	const targetType = resolveType(paramType, service, { spread, union: true })
+	return dtsParameter(name, targetType, {
+		spread,
+		optional,
+		jsDocComment: doc,
+		service
+	})
+}
+
+const operationToFunction = (operation, service) => {
+	const {
+		name,
+		params,
+		ret: { type: returnType } = {},
+		docs: { summary } = {}
+	} = operation
+	const targetReturnType = resolveType(returnType, service, {
+		intersection: true
+	})
+	const documentationGenerator = getDocumentationGenerator()
+	const parameters = params.map(p => operationParamToParameters(p, service))
+	const jsDocComment = documentationGenerator({
+		summary: summary,
+		service: name,
+		member: name
+	})
+
+	return dtsFunction(name, parameters, targetReturnType, {
+		jsDocComment,
+		service
+	})
+}
+
+const messageMemberToProperty = (member, service) => {
+	const { name, type, optional, doc: jsDocComment } = member
+	const resolvedType = resolveType(type, service, { union: true })
+	return dtsProperty(name, resolvedType, { optional, jsDocComment, service })
+}
+
+const messageToType = (message, service) => {
+	const {
+		name,
+		members = [],
+		enum: messageEnum,
+		docs: { summary: jsDocComment } = {}
+	} = message
+	if (!isEmpty_(messageEnum)) {
+		return dtsEnum(name, messageEnum, { jsDocComment })
+	} else {
+		const properties = members.map(member =>
+			messageMemberToProperty(member, service)
+		)
+		return dtsObjectTypeAlias(name, properties, { jsDocComment })
+	}
+}
+
+const callbackToFunctionType = (callback, service) => {
+	const {
+		name,
+		params,
+		ret: { type: returnType } = {},
+		docs: { summary: jsDocComment } = {}
+	} = callback
+	const parameters = params.map(param =>
+		operationParamToParameters(param, service)
+	)
+
+	const resolvedReturnType = resolveType(returnType, service, { union: true })
+
+	return dtsFunctionTypeAlias(name, parameters, resolvedReturnType, {
+		jsDocComment,
+		service
+	})
+}
+
+const propertyToProperty = (property, service) => {
```

Comment:
like we said, it would be better to name it `propertyToDtsProperty` or `serviceJsonPropertyToDtsProperty`
same for all the other converters.

In order to not repeat yourself in each function name you can also export these functions as part of an object, and then use them:
`serviceJsonToDtsCoverter.convertProperty()`